addon/mixins/store.js

```diff
@@ -223,10 +224,11 @@ const StoreMixin = {
           let originalRemove = proto.remove;
           proto.__originalRemove = originalRemove;
           proto.remove = function remove(internalModel) {
-            delete store._globalM3Cache[internalModel.id];
+            delete internalModel.store._globalM3Cache[internalModel.id];
             return originalRemove.apply(this, arguments);
```

Comment:
Looks like we are still closing over `originalRemove`, thoughts on using `this.__originalRemove` instead?

Reply:
Looks like I forgot to submit this, and it has been addressed. Awsome!